internal/compose/convert.go

```diff
@@ -79,18 +101,17 @@ func ConvertSpec(spec *score.Workload) (*compose.Project, ExternalVariables, err
 			Volumes:     volumes,
 		}
 
-		var proj = compose.Project{
-			Services: compose.Services{
-				svc,
-			},
+		if len(spec.Containers) > 1 {
+			// if we have more than 1 container then namespace them with the container name
+			svc.Name += "-" + containerName
+			// if we are not the "first" service, then inherit the network from the first service
+			if len(project.Services) > 0 {
+				svc.Ports = nil
+				svc.NetworkMode = "service:" + project.Services[0].Name
```

Comment:
The [Compose Spec](https://github.com/compose-spec/compose-spec/blob/master/05-services.md#network_mode) states:
> `service:{name}`: Gives the containers access to the specified service only.

Doesn't this mean that the container will not longer be able to access other services in the compose file? (e.g. database)

Reply:
The implementation in docker is that the network namespace is inherited https://docs.docker.com/network/#container-networks, so the not-first containers get no control over networking properties but they get all the same hostname, mac address, routing tables, conntrack tables, etc.

```
        "HostConfig": {
            "NetworkMode": "container:792f7ce24ee1c0ba05ea53dcb05d29a0acca82aeedc1cf5492888780b518869a",
```

I'm not sure why the compose spec says "gives access". That doesn't seem correct since I can launch too `debian` containers with Score like this can verify that both get the same `ifconfig` output and mac while also being able to curl google.com successfully.

I also verified that both can listen on non-overlapping ports and receive traffic just fine.

Reply:
Notably it also correctly recreates container-B when container-A is recreated.